src/linear_algebra/finite_dimensional.lean

```diff
@@ -558,6 +564,8 @@ basis.of_repr
   basis_singleton ι h v hv i = v :=
 by { cases unique.uniq ‹unique ι› i, simp [basis_singleton], }
 
+attribute [irreducible] basis_singleton
```

Comment:
I don't really see the appeal here. This one genuinely has interesting definitional content, unlike all the fintype instances above.